src/sage/combinat/designs/difference_family.py

```diff
@@ -1616,41 +1643,81 @@ def is_supplementary_difference_set(Ks, v, lmbda):
 
     - ``lmbda`` -- integer, the parameter `\lambda` of the supplementary difference sets.
 
+    - ``G`` -- a group of order `v`.
+
+    - ``verbose`` -- boolean (default False). If true the function will be verbose
+      when the sets do not satisfy the contraints.
+
     EXAMPLES::
 
         sage: from sage.combinat.designs.difference_family import supplementary_difference_set_from_rel_diff_set, is_supplementary_difference_set
-        sage: S1, S2, S3, S4 = supplementary_difference_set_from_rel_diff_set(17)
-        sage: is_supplementary_difference_set([S1, S2, S3, S4], 16, 16)
+        sage: G, [S1, S2, S3, S4] = supplementary_difference_set_from_rel_diff_set(17)
+        sage: is_supplementary_difference_set([S1, S2, S3, S4], lmbda=16, G=G)
         True
-        sage: is_supplementary_difference_set([S1, S2, S3, S4], 16, 14)
+
+    The parameter ``v`` can be given instead of ``G``::
+
+        sage: is_supplementary_difference_set([S1, S2, S3, S4], v=16, lmbda=16)
+        True
+        sage: is_supplementary_difference_set([S1, S2, S3, S4], v=20, lmbda=16)
+        False
+
+    If ``verbose`` is true, the function will be verbose::
+
+        sage: is_supplementary_difference_set([S1, S2, S3, S4], lmbda=14, G=G, verbose=True)
+        Number of pairs with difference (1) is 16, but lambda is 14
         False
-        sage: is_supplementary_difference_set([S1, S2, S3, S4], 20, 16)
+
+    TESTS::
+
+        sage: is_supplementary_difference_set([[1], [1]], lmbda=0, G=Zmod(3))
+        True
+        sage: is_supplementary_difference_set([S1, S2, S3, S4], v=17, lmbda=16, G=G)
         False
+        sage: is_supplementary_difference_set([S1, S2, S3, S4], G=G)
+        True
+        sage: is_supplementary_difference_set([S1, S2, S3, S4], lmbda=16)
+        Traceback (most recent call last):
+        ...
+        ValueError: One of G or v must be specified.
 
     .. SEEALSO::
 
         :func:`supplementary_difference_set_from_rel_diff_set`
     """
-    from sage.groups.additive_abelian.additive_abelian_group import AdditiveAbelianGroup
 
-    G = AdditiveAbelianGroup([v])
-    differences_counter = {}
+    if G is None and v is None:
+        raise ValueError('One of G or v must be specified.')
```

Comment:
```suggestion
        raise ValueError('one of G or v must be specified')
```
We follow Python conventions for error messages of initial lowercase without a period.